### DISCUSSION TITLE
permanent redirect in Next JS 13

### DISCUSSION THREAD

**Discussion by MahmoudBakr22** [vercel/next.js#49853] [App Router]
in Next JS 12 if you want to redirect you used to return an object from getServerSideProps that holds the distination path and if it is a permanent redirect or not

```
  return {
    redirect: {
      destination: "/",
      permanent : true
    },
  };
```
now in Next js 13 we should use redirect() in server components but this redirect() function does not accept any parameters other than the destination https://nextjs.org/docs/app/api-reference/functions/redirect

**Comment by JurajKavka** [comment]
Hello, any answers here? I need to do permanent redirect from app router and I would like to know hot o do it. My use case is to read data from API and based on them decide, if redirect is temporary or permanent.

**Comment by JurajKavka** [comment]
it is impelemented in current canary release 👍 

https://nextjs.org/docs/app/api-reference/functions/permanentRedirect

I'm using it successfully in `generateMetadata` function, that send header before streaming starts ...

**Reply by QingjiaTsang** [reply]
nextjs on v13.4.19

how come I use it like the example in the doc at the server component for permanent redirect, but it fails and says "Module '"next/navigation"' has no exported member 'permanentRedirect'.ts(2305)"

here are the codes below:
![image](https://github.com/vercel/next.js/assets/39949421/c6c0897c-fb53-4a36-93d6-f5d27ab5033e)

error:
![image](https://github.com/vercel/next.js/assets/39949421/9b547552-c29b-4a6a-8c53-76fae84d6a6a)
